Agents PoP

toschoosqd opened the pull request:
Outline of solution for "Agents" project.

toschoosqd committed:
[agents-pop] first draft

toschoosqd committed:
Update 2025-11-27-agents-pop-1.md

toschoosqd committed:
[agents-pop] diagram reviewed

mo4islona commented:
This is the winner of our hackathon at EthGlobal.
https://ethglobal.com/showcase/data-context-market-02545

It is worth careful review.

@toschoosqd 
We can reuse/borrow many ideas/implementations from this prototype (UX/MCP part)

kalabukdima commented:
Sounds good! Let's then define the next deliverable milestone to be "add at least one custom dataset queriable with SQL"

toschoosqd committed:
Previous work, notes on missing links in 2025-11-27-agents-pop-1.md

kalabukdima commented on sql-research/2025-11-27-agents-pop-1.md:
```diff
@@ -0,0 +1,146 @@
+
+- Which models / services?
+
+- How to provide metadata and other contextual data (e.g.: vector DB)?
```
As I understand it, vector databases allow storing data with binary vectors as keys and performing searches like "give me K closest entries to this vector together with their payloads". Storing binary data is easy, the main question is — can DuckDB or polars-based query engine perform searches like this?

toschoosqd replied:
I'm not yet sure about the architecture we would need and its implementation details; from my understanding, it would not be the query engine that communicates with the vector database, but the agent. But that certainly depends on what kind of context information we want to store and which role it has overall. In another project we used Qdrant, which exposes an endpoint to interact with the vector database itself. That made it quite easy to insert and retrieve data.

kalabukdima commented on sql-research/2025-11-27-agents-pop-1.md:
```diff
@@ -0,0 +1,146 @@
+
+- Integration: do we need integration with network (for resource usage)?
+
+- Integration with CoinBase? SQD?
```
I think if we integrate x402 payments through the portal (out of scope of this task), everything else will be straightforward. I assume it will be enough to write accurate API documentation and point the AI agent to this documentation, the API, and the metadata description to figure out how to access data.

kalabukdima commented on sql-research/2025-11-27-agents-pop-1.md:
```diff
@@ -0,0 +1,146 @@
+
+Tasks to implement this solution:
+
+1. Define relevant / exciting data
```
A couple of candidates to be onboarded first:
- All ERC20 transfers on ethereum-mainnet
- ABI-decoded transactions — we can only decode a subset of known contracts, so it's a harder one
- Metadata of ERC20 tokens as they are minted (decimals, ticker, etc)

toschoosqd committed:
Enhance roadmap with comments and metadata tasks

Added comments and proposed a roadmap for integrating enriched datasets and metadata definition.
